inc/scroom/layeroperations.hh

```diff
@@ -29,6 +30,16 @@ public:
                     Scroom::Utils::Stuff cache);
 };
 
+class PipetteCommonOperations : public CommonOperations, public PipetteLayerOperations
+{
+public:
+  virtual ~PipetteCommonOperations()
+  {}
+
+  PipetteLayerOperations::PipetteColor sumPixelValues(Scroom::Utils::Rectangle<int> area, const ConstTile::Ptr tile, int bps, int spp);
```

Comment:
But... but...

It is the job of the `LayerOperations` object to know the `spp` and `bps` values. Otherwise, it would never be able to do such things as `redraw()` and `reduce()`.

Hence, it should not be necessary to pass them in here.

Reply:
I guess we can add both `spp` and `bps` to `PipetteCommonOperations`  and define getters `getSpp` and `getBps` which each operation which supports pipette will implement?

Alternative would be to make them public in `PipetteCommonOperations` . If we split up the method `sumPixelValues` for CMYK vs RGB, we could get away with a hardcoded `spp` value and compute the `bps` from the `bpp`.

Reply:
I'm not a big fan of getters. I guess I'd make them protected. For now, we're only talking about PipetteCommonOperations, and since that's the class that implements `sumPixelValues()`, you might even make them private and not have any getters.